torch/_inductor/select_algorithm.py

```diff
@@ -1665,7 +1665,9 @@ def benchmark_example_value(node):
         if isinstance(node, ir.Layout):
             node = ir.Buffer("fake", node)
         # triton templates want the base tensor.
-        if isinstance(node, ir.BaseView):
+        if isinstance(node, ir.BaseView) and not (
+            isinstance(node, ir.ReinterpretView) and node.layout.device.type == "cpu"
```

Comment:
Why this?

Reply:
In the following code https://github.com/pytorch/pytorch/blob/b408591b5380d6b856e0a4ce32cf386c56660c54/torch/_inductor/select_algorithm.py#L1670-L1682, we will use the `size/stride` from this `node` to generate inputs to do benchmark. Take the case in this PR as example: 
```
ReinterpretView(
  StorageBox(
    ConstantBuffer(name='L__self___mlp_0_weight', layout=FixedLayout('cpu', torch.float32, size=[64, 128], stride=[128, 1]))
  ),
  FixedLayout('cpu', torch.float32, size=[128, 64], stride=[1, 128]),
  origins=OrderedSet([addmm])
)
```
with or without `unwrap_view`, we will get different `size/stride`. And I think we need the `size/stride` after view as inputs to the gemm template for this case.

cc @constroy who adds the comment `# triton templates want the base tensor.` in PR: https://github.com/pytorch/pytorch/pull/105315. Do you know why triton templates need to `unwrap_view`?

Reply:
Same question - What's the problem without `unwrap_view()` call here? Should we build the example value according to the actual size/stride of the node?

Reply:
If some arguments passed to the triton template are views from one base tensor, we can de-duplicate them by `unwrap_view`. And it should result in better code.

Reply:
If the view will change the size and stride, it will lead to a different tensor. Per my understanding, we should use the tensor before `unwrap_view` for this case. And it causes difference between `example_inputs` and `example_inputs_extern` specifically:

- We use the tensor size after `unwrap_view` for example inputs as https://github.com/pytorch/pytorch/blob/b408591b5380d6b856e0a4ce32cf386c56660c54/torch/_inductor/select_algorithm.py#L1670-L1682
- And tensor size before  `unwrap_view` for `example_inputs_extern` https://github.com/pytorch/pytorch/blob/b408591b5380d6b856e0a4ce32cf386c56660c54/torch/_inductor/select_algorithm.py#L1431-L1452

Reply:
They are meant to be different here. Triton templates can make up the actual stride/offset inside its kernel function.
https://github.com/pytorch/pytorch/blob/b408591b5380d6b856e0a4ce32cf386c56660c54/torch/_inductor/select_algorithm.py#L1487-L1492